File: abc/abc174/abc174_c.fs
// https://atcoder.jp/contests/abc174/submissions/15652998

let K = stdin.ReadLine() |> int

let rec solver q i =
    if q = 0 then i
    elif K < i then -1
    else solver ((10 * q + 7) % K) (i + 1)

solver (7 % K) 1 |> stdout.WriteLine

File: abc/abc174/abc174_c.cs
// https://atcoder.jp/contests/abc174/submissions/15653049

using System;

class Program
{
    static void Main()
    {
        var K = int.Parse(Console.ReadLine());
        var q = 7 % K;
        var result = 1;

        while (true)
        {
            if (q == 0)
            {
                break;
            }
            if (K < result)
            {
                result = -1;
                break;
            }
            q = (10 * q + 7) % K;
            result++;
        }
        Console.WriteLine(result);
    }
}
